### DISCUSSION TITLE
Jandex vs beans.xml : which is better to use?

### DISCUSSION THREAD

**Discussion by daniel-alonso-sanchez** [quarkusio/quarkus#38503] [Q&A] [answered]
We are having some discussions at my current company about which approach is more advisable, whether to use beans.xml or Jandex when facilitating the discovery of beans in libraries developed for Quarkus.

In terms of startup times, is one of the two better than the other? And regarding other aspects?

Thanks a ton in advance!

**Comment by Ladicek** [comment, marked as answer]
There's no difference in startup time or runtime behavior in general. If you include `beans.xml`, Quarkus will build a Jandex index automatically, whereas if you include a Jandex index yourself, Quarkus will use it and won't bother indexing on its own. So it is really a matter of taste. (Creating an index is plenty fast, so if you feel like pre-indexing your libraries could speedup the Quarkus build process, you're thinking in the right direction, but unless you have really massive libraries, you shouldn't notice a difference.)

My personal opinion is that you should use `beans.xml` in all archives that you want to be bean archives, and if you do have `beans.xml`, you don't have to care about configuring the Jandex Maven/Gradle plugin.

**Reply by daniel-alonso-sanchez** [reply]
Thank you so much for your time & your answer, @Ladicek !
